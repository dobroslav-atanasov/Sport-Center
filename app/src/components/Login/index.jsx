import React from 'react';
import { Link, Redirect } from 'react-router-dom';
import userService from '../../services/userService';

class Login extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: '',
            errors: '',
            // isRedirect: false
        };
    };

    submitHandler = (event) => {
        event.preventDefault();
        const data = this.state;
        userService.login(data).then(res => {
            if (res === 'Invalid username or password!') {
                this.setState({ errors: res });
            } else {
                this.props.history.push('/');
                // this.setState({ isRedirect: true });
            }
        });
    };

    changeHandler = (event) => {
        event.preventDefault();
        const { name, value } = event.target;
        this.setState({ [name]: value });
    }

    render() {
        return (
            // this.state.isRedirect ? <Redirect to="/" /> :
                <div className="container">
                    <br />
                    <div className="card bg-light">
                        <article className="card-body mx-auto">
                            <h3 className="card-title mt-3 text-center">Log In</h3>
                            <hr />
                            <form onSubmit={this.submitHandler}>
                                {/* USERNAME */}
                                <div className="form-group input-group">
                                    <div className="input-group-prepend">
                                        <span className="input-group-text"><i className="fa fa-user"></i></span>
                                    </div>
                                    <input type="text" className="form-control" name="username" onChange={this.changeHandler} placeholder="Username" />
                                </div>

                                {/* PASSWORD */}
                                <div className="form-group input-group">
                                    <div className="input-group-prepend">
                                        <span className="input-group-text"> <i className="fa fa-lock"></i> </span>
                                    </div>
                                    <input type="password" className="form-control" name="password" onChange={this.changeHandler} placeholder="Password" />
                                </div>
                                {this.state.errors.length > 0 && <div className="alert alert-warning">{this.state.errors}</div>}

                                {/* SUBMIT */}
                                <div class="form-group">
                                    <button type="submit" className="btn btn-danger btn-block">Log In</button>
                                </div>
                                <p class="text-center">Don't have account? <Link to="/register">Create Account</Link></p>
                            </form>
                        </article>
                    </div>
                </div>
        )
    }
}

export default Login;