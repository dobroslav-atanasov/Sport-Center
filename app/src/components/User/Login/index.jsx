import React from 'react';
import { Link } from 'react-router-dom';
import userService from '../../../services/userService';
import constants from '../../../constants/constants';

class Login extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: '',
            errors: ''
        };
    };

    submitHandler = (event) => {
        event.preventDefault();
        const data = this.state;
        userService.login(data).then(res => {
            if (res === constants.login.INVALID_USERNAME_PASSWORD) {
                this.setState({ errors: res });
            } else {
                this.props.history.push('/');
            }
        });
    };

    changeHandler = (event) => {
        event.preventDefault();
        const { name, value } = event.target;
        this.setState({ [name]: value });
    }

    render() {
        const { errors } = this.state;
        return (
            <div className="container" style={{ marginTop: 30, marginBottom: 50 }}>
                <div className="card bg-light">
                    <article className="card-body mx-auto">
                        <h3 className="card-title mt-3 text-center">Log In</h3>
                        <hr />
                        <form onSubmit={this.submitHandler}>
                            {/* USERNAME */}
                            <div className="form-group input-group">
                                <div className="input-group-prepend">
                                    <span className="input-group-text"><i className="fa fa-user"></i></span>
                                </div>
                                <input id="username" type="text" className="form-control" name="username" onChange={this.changeHandler} placeholder="Username" />
                            </div>

                            {/* PASSWORD */}
                            <div className="form-group input-group">
                                <div className="input-group-prepend">
                                    <span className="input-group-text"> <i className="fa fa-lock"></i> </span>
                                </div>
                                <input id="password" type="password" className="form-control" name="password" onChange={this.changeHandler} placeholder="Password" />
                            </div>
                            {errors.length > 0 && <div className="alert alert-danger"><i className="fa fa-exclamation-triangle"></i> {errors}</div>}

                            {/* SUBMIT */}
                            <div className="form-group">
                                <button type="submit" className="btn btn-danger btn-block">Log In</button>
                            </div>
                            <p className="text-center">Don't have account? <Link to="/register">Create Account</Link></p>
                        </form>
                    </article>
                </div>
            </div>
        );
    };
};

export default Login;