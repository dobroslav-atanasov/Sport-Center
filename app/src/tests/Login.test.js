import React from 'react';
import Login from '../components/User/Login';
import renderer from 'react-test-renderer';

jest.mock('react-router-dom', () => ({
    Link: "Link"
}));

describe('Login component test', () => {
    test('Should render component', () => {
        const component = renderer.create(<Login />);

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Should have button', () => {
        const component = renderer.create(<Login />);
        const root = component.root;
        const element = root.findByType('button').children.length;
        expect(element).toEqual(1);
    });

    test('Should have button with text Log In', () => {
        const component = renderer.create(<Login />);
        const root = component.root;
        const result = root.findByType('button').children[0] === 'Log In';
        expect(result).toEqual(true);
    });

    // test('Should have button with name Log In', () => {
    //     const component = renderer.create(<Login />);
    //     const instance = component.root;

    //     const result = instance.find((el) => el.type === 'button');
    //     expect(result).toBe(true);
    // });

    // const wrapper = shallow(<Login />);

    // it('Should have input for username', () => {
    //     expect(wrapper.find('input#username')).toHaveLength(1);
    // });

    // it('Username input should have empty state', () => {
    //     expect(wrapper.state('username')).toEqual('');
    // });

    // it('Should have input for password', () => {
    //     expect(wrapper.find('input#password')).toHaveLength(1);
    // });

    // it('Password input should have empty state', () => {
    //     expect(wrapper.state('password')).toEqual('');
    // });

    // it('Should have a button', () => {
    //     expect(wrapper.find('button').text()).toEqual('Log In');
    // });
});