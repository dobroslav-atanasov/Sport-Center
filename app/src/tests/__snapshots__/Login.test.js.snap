// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component test Should render component 1`] = `
<div
  className="container"
  style={
    Object {
      "marginBottom": 50,
      "marginTop": 30,
    }
  }
>
  <div
    className="card bg-light"
  >
    <article
      className="card-body mx-auto"
    >
      <h3
        className="card-title mt-3 text-center"
      >
        Log In
      </h3>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group input-group"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
            >
              <i
                className="fa fa-user"
              />
            </span>
          </div>
          <input
            className="form-control"
            id="username"
            name="username"
            onChange={[Function]}
            placeholder="Username"
            type="text"
          />
        </div>
        <div
          className="form-group input-group"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
            >
               
              <i
                className="fa fa-lock"
              />
               
            </span>
          </div>
          <input
            className="form-control"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
        </div>
        <div
          className="form-group"
        >
          <button
            className="btn btn-danger btn-block"
            type="submit"
          >
            Log In
          </button>
        </div>
        <p
          className="text-center"
        >
          Don't have account? 
          <Link
            to="/register"
          >
            Create Account
          </Link>
        </p>
      </form>
    </article>
  </div>
</div>
`;
